src/index.ts

```diff
@@ -0,0 +1,158 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+const { getPromiseDetails } = process.binding('util');
+
+/**
+ * The class for deep checking Types
+ */
+export default class Type {
+
+	/**
+	 * The value to generate a deep Type of
+	 * @type {*}
```

Comment:
All type information from TSDocs should be removed, since they are already available in TypeScript.